### ISSUE TITLE
No-traceback "production mode"

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#3834]
**Describe exactly what you would like to see in an upcoming release**

As discussed in the Feb 2020 workshop, security people see tracebacks in production as a security issue, because they reveal information about the code base and system (paths) that users do not need to know.

(The fact that users on the system can easily determine the location of, and read, the Python source files is evidently beside the point :thinking: ).

So, we should provide a global config option to suppress all tracebacks in a secure production context.

**Pull requests welcome!**

**Comment by oliver-sanders** [comment]
Might be a bit difficult to suppress traceback in errors from external systems.

**Comment by hjoliver** [comment]
> Might be a bit difficult to suppress traceback in errors from external systems.

What do you mean by "external systems"?  (Stuff executed in task jobs, or other programs used by Cylc such as ssh and rsync?)

**Comment by oliver-sanders** [comment]
Yes, ssh, bash, etc, when we log stderr when a job-submit or remote-init fails we could capture traceback from non-cylc things.

**Comment by hjoliver** [comment]
I think we only need to worry about Cylc source tracebacks.
